src/dynamic_boundary_conditions/river/river_data_from_niwa.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+"""
+Fetch REC data in New Zealand from NIWA using the ArcGIS REST API.
+"""
+
+import logging
+from typing import Tuple, List, Dict, Union
+import asyncio
+
+import aiohttp
+import requests
+import geopandas as gpd
+import pandas as pd
+from shapely.geometry import LineString
+from sqlalchemy.engine import Engine
+
+from src.digitaltwin.utils import get_nz_boundary
+
+log = logging.getLogger(__name__)
+
+# URL for retrieving REC data from NIWA using the ArcGIS REST API
+REC_API_URL = "https://gis.niwa.co.nz/server/rest/services/HYDRO/Flood_Statistics_Henderson_Collins_V2/MapServer/2"
+
+
+def get_feature_layer_record_counts(url: str = REC_API_URL) -> Tuple[int, int]:
+    """
+    Retrieves the maximum and total record counts from the REC feature layer.
+
+    Parameters
+    ----------
+    url : str = REC_API_URL
+        The URL of the REC feature layer. Defaults to 'REC_API_URL'.
+
+    Returns
+    -------
+    Tuple[int, int]
+        A tuple of integers containing the maximum record count and the total record count.
+        - max_record_count (int): The maximum number of records that will be returned per query.
+        - total_record_count (int): The total number of records available in the feature layer.
```

Comment:
Don't worry about this now but I like to use `NamedTuple` or `dataclass` for these kind of returns so the developers using it can see what each of the values represent more clearly. `NamedTuple` would work well here

Reply:
I've updated this to a `NamedTuple`. Let me know if there are additional changes you'd like me to make to convert it to a `NamedTuple`.

Reply:
I think its okay for now like I said. You have used them correctly here. It would be good to use them in the future whenever you can.